src/Compilers/Core/Portable/PEWriter/PeWriter.cs

```diff
@@ -264,6 +276,9 @@ internal static class PeWriter
                 throw new PeWritingException(e);
             }
 
+            peBlob.Free();
+            managedResourceBuilder?.AssertFreed();
```

Comment:
Isn't this a race condition? The builder could have been lent out again by this point.

Reply:
Good catch, that is a race condition. Hmm ... I really wanted to have an assert here given the interesting relationship with `Count == 0` described above. Not seeing how I can do that though. May just have to back out the assert code and just trust the comment here.